OpenVINO 2024 not working properly

### System Information

OpenCV Version: 4.10.0
Operating System / Platform: Windows 11 x64
Compiler & compiler version: VS 2022

### Detailed description

With the 4.10.0 release, support for OpenVINO 2024 is included. Compiling and linking against OpenVINO 2024 works, however activating at runtime results in runtime error.s

### Steps to reproduce

1. Compile OpenVINO 2024. We used 2024.2.0, only using the CPU plugin and compiled it as static libraries.
2. Compile OpenCV using OpenVINO, except those necessary for OpenVINO, further CMake configurations are not required.
3. Thereafter, run the following code:

```python
import numpy as np
import cv2 as cv

# model is available in models/image_classification_mobilenet inside OpenCV's zoo repository
model = cv.dnn.readNet('image_classification_mobilenetv2_2022apr.onnx')

# use OpenVINO backend (no error if DNN_BACKEND_OPENCV is used instead)
model.setPreferableBackend(cv.dnn.DNN_BACKEND_INFERENCE_ENGINE)

# forward zero image
model.setInput(np.zeros((1,3,224,224), dtype=np.float32))
model.forward()
```

Here, `forward()` terminates with the following error:

> error: OpenCV(4.10.0) opencv\modules\dnn\src\ie_ngraph.cpp:558: error: (-215:Assertion failed) blobIt != allBlobs.end() in function 'cv::dnn::InfEngineNgraphNet::forward'

As both, OpenCV's backend as well as OpenVINO 2023 work fine, the issue is caused by using OpenVINO 2024.

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

Spend a couple hours after I had updated my toolchain to Openvino 2024.3. I too have a:
`OpenCV(4.10.0-dev) D:\Dev\opencv\modules\dnn\src\ie_ngraph.cpp:558: error: (-215:Assertion failed) blobIt != allBlobs.end() in function 'cv::dnn::InfEngineNgraphNet::forward'`
Is there a known workaround?

--- a/modules/dnn/src/ie_ngraph.cpp
+++ b/modules/dnn/src/ie_ngraph.cpp
@@ -162,7 +162,9 @@ void InfEngineNgraphNet::createNet(Target targetId) {
             CV_LOG_DEBUG(NULL, "DNN/NGRAPH: Add 'Result' output: " << output_node_it->first);
             CV_Assert(output_node_it->second);
             auto out = std::make_shared<ov::op::v0::Result>(output_node_it->second->node);
-            out->set_friendly_name(output_node_it->first + (output_node_it->second->node.get_node()->get_output_size() == 1 ? "" : ".0"));
+            std::string name = output_node_it->first + (output_node_it->second->node.get_node()->get_output_size() == 1 ? "" : ".0")
+            CV_LOG_DEBUG(NULL, "DNN-IE: Change friendly name from " << out->get_friendly_name() << " to " << name);
+            out->set_friendly_name(name);
             outs.push_back(out);
         }
         CV_Assert_N(!inputs_vec.empty(), !outs.empty());
@@ -546,16 +548,22 @@ void InfEngineNgraphNet::forward(const std::vector<Ptr<BackendWrapper> >& outBlo
         {
             const std::string& name = it.get_node()->get_friendly_name();
             auto blobIt = allBlobs.find(name);
-            CV_Assert(blobIt != allBlobs.end());
+            if (blobIt == allBlobs.end())
+            {
+                CV_Error(Error::StsAssert, format("Input blob with name %s not found", name.c_str()));
+            }
             reqWrapper->req.set_input_tensor(i++, isAsync ? copyBlob(blobIt->second) : blobIt->second);
         }
 
         i = 0;
-        for (const auto& it : netExec.outputs())
+        for (const auto& it : cnn->outputs())  // Starts from OpenVINO 2024 CompiledModel changes output frindly names
         {
             const std::string& name = it.get_node()->get_friendly_name();
             auto blobIt = allBlobs.find(name);
-            CV_Assert(blobIt != allBlobs.end());
+            if (blobIt == allBlobs.end())
+            {
+                CV_Error(Error::StsAssert, format("Output blob with name %s not found", name.c_str()));
+            }
             reqWrapper->req.set_output_tensor(i++, isAsync ? copyBlob(blobIt->second) : blobIt->second);
         }
 

